pkg/provider/authentik/authentik.go

```diff
@@ -0,0 +1,284 @@
+	return true, next, err
+}
+
+// doPostQuery For all data setting operations
+func (kc *Client) doPostQuery(ctx *authentikContext, payload *authentikPayload) (string, error) {
+	data, err := getLoginJSON(ctx.loginDetails, payload)
+	if err != nil {
+		return "", err
+	}
+
+	logger.Debug("[POST]", ctx.loginDetails.URL)
+	res, err := kc.client.Post(ctx.loginDetails.URL, "application/json", bytes.NewReader(data))
+	if err != nil {
+		return "", err
+	}
+	if res.StatusCode == http.StatusOK {
+		var payload *authentikPayload
+		payload, err = parseResponsePayload(res)
+		if err != nil {
+			return "", err
+		}
+
+		var errMsg string
+		if len(payload.Errors) > 0 {
+			errMsg = prepareErrors(payload.Component, payload.Errors)
+		} else {
+			errMsg = "Unexpected"
+		}
+
+		return "", errors.New(errMsg)
+	}
+	loc, err := res.Location()
+	return loc.String(), err
+}
+
+// getLoginJSON Generate the login json
+func getLoginJSON(loginDetails *creds.LoginDetails, payload *authentikPayload) ([]byte, error) {
+	component := payload.Component
+	m := map[string]string{
+		"component": component,
+	}
+	switch component {
+	case "ak-stage-identification":
+		m["uid_field"] = loginDetails.Username
+		if payload.HasPassowrdField {
+			m["password"] = loginDetails.Password
+		}
+	case "ak-stage-password":
+		m["password"] = loginDetails.Password
+	default:
+		return []byte(""), errors.New("unknown component: " + component)
+	}
+
+	return json.Marshal(m)
+}
+
+// queryNextURL Get the next api url
+func queryNextURL(u string) (string, error) {
+	next, err := url.Parse(u)
+	if err != nil {
+		return "", errors.New("Invalid url")
+	}
+
+	result := strings.Split(next.Path, "/")
+	flow := result[len(result)-2]
+	return fmt.Sprintf("%s://%s/api/v3/flows/executor/%s/?query=%s", next.Scheme, next.Host, flow, url.QueryEscape(next.RawQuery)), nil
+}
+
+// getFieldName Get name of component
+func getFieldName(component string) (string, error) {
+	prefix := "ak-stage-"
+	if strings.Index(component, prefix) != 0 {
+		return "", errors.New("")
+	}
+	s := strings.Split(component, "ak-stage-")
+	return s[len(s)-1], nil
+}
+
+// prepareErrors Transform errors to string
+func prepareErrors(component string, errs map[string][]map[string]string) string {
+	field, err := getFieldName(component)
+	if err != nil {
+		return "Invalid component"
+	}
+
+	key := "non_field_errors"
+	if field == "password" {
+		key = "password"
+	}
+	msgs := make([]string, 0, len(errs[key]))
+	for _, err := range errs[key] {
+		msgs = append(msgs, fmt.Sprintf("%s %s: %s", field, err["code"], err["string"]))
+	}
+	return strings.Join(msgs, "; ")
+}
+
+// parseResponsePayload Parse response from authentik api
+func parseResponsePayload(res *http.Response) (*authentikPayload, error) {
+	var payload authentikPayload
+	defer res.Body.Close()
+	b, err := ioutil.ReadAll(res.Body)
+	err = json.Unmarshal(b, &payload)
+	if err != nil {
+		return nil, err
+	}
+
+	return &payload, nil
+}
+
+func aaa(a int) int {
```

Comment:
Is this intentional? Could we find a better function name?

Reply:
```
  Error: `aaa` is unused (deadcode)
  Error: File is not `goimports`-ed (goimports)
  Error: ineffectual assignment to err (ineffassign)
  Error: SA1019: "io/ioutil" has been deprecated since Go 1.19: As of Go 1.16, the same functionality is now
```

Reply:
It's a mistake, removed